internal/pkg/service/appsproxy/dns/client.go

```diff
@@ -0,0 +1,88 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const DNSReadTimeout = 2 * time.Second
+
+const DNSWriteTimeout = 2 * time.Second
+
+type Client struct {
+	client    *dns.Client
+	dnsServer string
+}
+
+func NewClient() (*Client, error) {
+	// Dialer
+	dialer := newDialer()
+
+	// DNS client
+	client := &dns.Client{
+		Net:          "udp",
+		Dialer:       dialer,
+		DialTimeout:  DialTimeout,
+		ReadTimeout:  DNSReadTimeout,
+		WriteTimeout: DNSWriteTimeout,
+	}
+
+	// Parse DNS config
+	dnsCfgFile := "/etc/resolv.conf"
+	dnsCfg, err := dns.ClientConfigFromFile(dnsCfgFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(dnsCfg.Servers) == 0 {
+		return nil, errors.Errorf(`no DNS server found in "%s"`, dnsCfgFile)
+	}
+
+	// Get DNS server
+	dnsServer := dnsCfg.Servers[0]
+	if !strings.Contains(dnsServer, ":") {
+		dnsServer += ":53"
+	}
+
+	c := &Client{
+		client:    client,
+		dnsServer: dnsServer,
+	}
+
+	return c, nil
+}
+
+func (c *Client) Resolve(ctx context.Context, host string) (string, error) {
+	msg := &dns.Msg{}
+	msg.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	msg.Authoritative = true
+	msg.RecursionAvailable = false
+	msg.RecursionDesired = false
+
```

Comment:
Please add comment why recursion is disabled:
- It skip caching - by a side effect.
- We want to know if some service pod available in k8s, we don't need to go recursively to other servers.